src/gui/qgsbrowserdockwidget_p.cpp

```diff
@@ -154,6 +154,20 @@ void QgsBrowserLayerProperties::setItem( QgsDataItem *item )
       delete layer;
     }
   }
+  else if ( type == QgsMapLayer::MeshLayer )
+  {
+    QgsDebugMsg( "creating mesh layer" );
+    QgsMeshLayer *layer = new QgsMeshLayer( layerItem->uri(), layerItem->uri(), layerItem->providerKey() );
```

Comment:
unique_ptr

Reply:
I would rather keep it this way, since both if ( type == QgsMapLayer::RasterLayer ) and else if ( type == QgsMapLayer::VectorLayer ) do it without unique_ptr

Reply:
Boy Scout that code then! Don't let new code be influenced by existing bad practice, instead fix the existing code and leave the codebase in a cleaner state then it was before.